Add SwiftPM Support

JoeMatt opened the pull request:
Added a Package.swift to support iOS, tvOS and macCatalyst.

Also supports tests.

No warnings or errors. Added some minor changes for deprecations and warnings such as `class` to `AnyObject`.

AppDevGuy reviewed (approved):
Looks good, thanks for adding.

AppDevGuy commented:
Build is failing because it's attempting to use an outdated version of Xcode. I will have to upgrade this package to get it working, might be a weekend thing.

JoeMatt commented:
> Build is failing because it's attempting to use an outdated version of Xcode. I will have to upgrade this package to get it working, might be a weekend thing.

I pushed an update that I think will fix that.

- Lowered SPM version to 5.3
- tvOS only compiles TextToSpeech, not SpeechToText
- Replace import `UIKit` with `Foundation`, and conditionally compile `UIImage* tag` support. (For possibly supporting macOS or other targets that don't have UIKit but have the other requirements)
- Loading bundle uses `Bundle.module` if `SWIFT_PACKAGE`.

JoeMatt commented:
Oh. I see you meant the gh workflow. I can take a look at that real quick.

AppDevGuy commented on OSSSpeechKit/Classes/OSSSpeechUtility.swift:
```diff
@@ -72,33 +73,39 @@ public class OSSSpeechUtility: NSObject {
-        return Bundle(url: resourcesBundleUrl)
-    }
+	/// Will return the Bundle for the SDK if it can be found.
+	static func getResourcesBundle() -> Bundle? {
```
Unless I am not mistaken, indentation here has increased, which is probably due to the `#if` statement, Xcode always messes up indentation when implementing those macros.

AppDevGuy commented on OSSSpeechKit/Classes/OSSVoice.swift:
```diff
@@ -226,12 +229,14 @@ public enum OSSVoiceEnum: String, CaseIterable {
     /// You can supply your own flag image, provided is has the same name (.rawValue) as the image in the pod assets.
     ///
     /// If no image is found in the application bundle, the image from the SDK bundle will be provided.
+	#if canImport(UIKit)
```
Indentation is incorrect on the `#if` here as well.

JoeMatt replied:
Fixing. I've seen it both ways though. Swift-format allows indenting though Xcode by default doens't indent them.

AppDevGuy reviewed (commented):
Great catches and effort on identifying the tvOS issue for the Swift Package. Just a couple of indentation issues.

JoeMatt commented:
Pushed updates.

- Fixed? xcode gh action
- Added swift pm gh action
- Added macOS support (this is to allow swift build/text from cli without having to remember how to tell it to use iOS, though it works on mac now too so why not).
- Fixed indenting issues
- Fixed broken force unwrap in test

JoeMatt commented:
One last push after SwiftLint autocorrect.

AppDevGuy commented:
Tests failed which I believe I know the reason why - the `let archiver = NSKeyedUnarchiver(forReadingWith: Data())` whic you replaced - I went down this rabbit hole 2 years or more ago now when it was first deprecated but it seems Apple still haven't fixed the issue.

AppDevGuy commented:
Is there any chance you can rebase this on `v0.3.3_develop` branch? If so, I will merge it in there, then fix any of the small issues and do a release.

JoeMatt commented:
> Is there any chance you can rebase this on `v0.3.3_develop` branch? If so, I will merge it in there, then fix any of the small issues and do a release.

Done

AppDevGuy commented:
Thanks @JoeMatt - I will fix up a couple of things now and aim to release a new version soon.
